### ISSUE TITLE
Maintain Input Values In State

### ISSUE THREAD

**Issue by micahalcorn** [OriginProtocol/origin-dapp#415]
The conversation textarea is an uncontrolled component because this was the easiest way to intercept the return key and force a form submission. As a consequence, the element's value is either maintained across all conversations until changed/submitted or [currently] it clears when a user toggles to a different conversation. If I type something in conversation A and then navigate to conversation B before submitting the form, my incomplete message should remain intact when I return to conversation A. There may be a broader concern of navigating completely away from messages/conversations (to listings, for example) and losing the textarea value. The same could apply to listing creation, which can easily be lost. Perhaps some forms should be controlled by redux state at the app level.

**Comment by jordajm** [comment]
@micahalcorn - I see how redux may be the right place to manage state for messaging but it might get somewhat messy to manage `react-jsonschema-form` state in redux. After some googling, it looks like it's possible, but it also looks like it can get kind of hacky and verbose. The main problem is that managing form state in redux is typically done with `onchange` events on form fields, and those events aren't easy to hook into with machine-generated form HTML.

**Comment by jordajm** [comment]
To prevent data loss from user navigation on the listing create form, we might want to add a message when the user tries to navigate away from the workflow that warns them that they may lose data if they don't complete the workflow.

**Comment by micahalcorn** [comment]
@jordajm yeah, this one is messy and probably not a priority. I think you're probably on the right track with a prompt.

**Comment by micahalcorn** [comment]
My instinct here was probably informed by my experience in Discord and also here on GitHub. I guess we could use `localstorage` and let people move freely about the app, but we don't need to do this anytime soon.
